lib/modules/manager/pep621/schema.ts

```diff
@@ -8,6 +8,46 @@ const DependencyRecordSchema = z
   .record(z.string(), z.array(z.string()))
   .optional();
 
+const PdmSchema = z.object({
+  'dev-dependencies': DependencyRecordSchema,
+  source: z
+    .array(
+      z.object({
+        url: z.string(),
+        name: z.string(),
+        verify_ssl: z.boolean().optional(),
+      }),
+    )
+    .optional(),
+});
+
+const HatchSchema = z.object({
+  envs: z
+    .record(
+      z.string(),
+      z
+        .object({
+          dependencies: DependencyListSchema,
+          'extra-dependencies': DependencyListSchema,
+        })
+        .optional(),
+    )
+    .optional(),
+});
+
+// https://docs.astral.sh/uv/concepts/dependencies/#dependency-sources
+const UvSource = z.object({
+  git: z.string().optional(),
+  path: z.string().optional(),
+  url: z.string().optional(),
+  workspace: z.boolean(z.literal(true)).optional(),
```

Comment:
Setting `workspace = false` currently errors out in uv:
```console
$ uv lock
Using Python 3.12.1
error: Failed to build: `foo @ file:///<REDACTED>`
  Caused by: Failed to parse entry for: `httpx`
  Caused by: `workspace = false` is not yet supported
```

If this ends up being supported, I'm not sure what this will actually do, so probably best to skip this behaviour as `invalid-dependency-specification`.

Reply:
I would allow booleans, otherwise the full fold parsing will fail.